snuba/subscriptions/data.py

```diff
@@ -173,28 +255,47 @@ def from_dict(cls, data: Mapping[str, Any]) -> LegacySubscriptionData:
         if not data.get("aggregations"):
             raise InvalidQueryException("No aggregation provided")
 
-        return LegacySubscriptionData(
-            project_id=data["project_id"],
-            conditions=data["conditions"],
-            aggregations=data["aggregations"],
-            time_window=timedelta(seconds=data["time_window"]),
-            resolution=timedelta(seconds=data["resolution"]),
+        legacy_subs_dict = {
+            "project_id": data["project_id"],
+            "conditions": data["conditions"],
+            "aggregations": data["aggregations"],
+            "time_window": timedelta(seconds=data["time_window"]),
+            "resolution": timedelta(seconds=data["resolution"]),
+        }
+        legacy_subs_dict = cls.update_sessions_subscriptions_with_session_params(
+            data_dict=legacy_subs_dict, data=data
         )
+        return LegacySubscriptionData(**legacy_subs_dict)
```

Comment:
@evanh Do you know the status of LegacySubscriptionData and if it can be removed? I think it would help to simplify this implementation if we do not have to support both types of subscription data.

Reply:
I wouldn't bother with updating anything Legacy related. It is not used in prod at all and the only reason it's not outright deleted is to support onprem/oss that are not migrated yet.